lib/rules/no-double-assignments.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @fileoverview Rule to catch accidental variable overwriting
+ * @author Denis Sokolov
+ * @copyright 2015 Denis Sokolov. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/**
+ * Return a node previous to the current
+ * @param {RuleContext} context The rule context
+ * @param {ASTNode} node The current Node
+ * @returns {ASTNode} The previous node
+ */
+function getPrevNode(context, node) {
+    var prevToken = context.getTokenBefore(node);
```

Comment:
These methods have been moved from [`context`](http://eslint.org/docs/developer-guide/working-with-rules#the-context-object) to [`SourceCode`](http://eslint.org/docs/developer-guide/working-with-rules#contextgetsourcecode). You can get `SourceCode` from `context.getSourceCode()` once and pass it in whenever you need it.